tools/common/Rewriter.cs

```diff
@@ -136,11 +150,10 @@ FieldDefinition AddPublicStaticField (TypeDefinition inType, string fieldName, T
 
 		void PatchClassPtrUsage (Dictionary<string, FieldDefinition> classMap)
 		{
-			foreach (var path in assembliesToPatch) {
-				using var stm = new FileStream (path, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
-				using var module = ModuleDefinition.ReadModule (stm);
+			foreach (var assem in assemblies) {
+				var module = assem.MainModule;
 				PatchClassPtrUsage (classMap, module);
-				module.Write (ToOutputFileName (path));
+				module.Write ();
```

Comment:
Don't write out the assembly, the linker will do that later.

However, it's important to tell the linker that the assembly was changed, and that's done by changing the action:

https://github.com/xamarin/xamarin-macios/blob/cb34bfe86cba310293d45004a5799c68e60901bd/tools/dotnet-linker/AppBundleRewriter.cs#L1092-L1094

https://github.com/xamarin/xamarin-macios/blob/cb34bfe86cba310293d45004a5799c68e60901bd/tools/dotnet-linker/AppBundleRewriter.cs#L1103

Note that you should only save the assembly if you actually modified it.